### PR TITLE
Parallel Evaluator Upgrade

### PR DESCRIPTION
**Pull request by thartland**
This PR is a work in progress but aims to add evaluators to take advantage of other embarrassingly parallel portions of a typical BO algorithm -- initial sampling and instantiating a local optimizer on a batch of random initial points.

### CODE REVIEW

**Comment by thartland** [comment]
This PR is ready for review.

**Comment by thartland** [comment]
BODriverEx does parallel evaluations of the initial samples used for the initial training set (see [here](https://github.com/LLNL/hiop/blob/hiopbbpy-v0.08-dev/src/Drivers/hiopbbpy/BODriverEX.py#L97)) and also uses a parallel (MPI) evaluator (see [here](https://github.com/LLNL/hiop/blob/hiopbbpy-v0.08-dev/src/Drivers/hiopbbpy/BODriverEX.py#L113)) for the multi-start local search to estimate the global optima of the acquisition function in the BO loop. This can used with parallel evaluations ([here](https://github.com/LLNL/hiop/blob/hiopbbpy-v0.08-dev/src/Drivers/hiopbbpy/BODriverEX.py#L112)) of the objective in a batched BO setting. Details in src/hiopbbpy/opt/boalgorithm.py

**Review by cnpetra — APPROVED** [review]

**Comment by thartland** [comment]
@nychiang reminding you of this PR.

File: src/hiopbbpy/opt/acquisition.py [hunk 1]
```diff
@@ -19,10 +19,16 @@ def __init__(self, gpsurrogate):
   # Abstract method to evaluate the acquisition function at x.
   def evaluate(self, x: np.ndarray) -> np.ndarray:
     raise NotImplementedError("Child class of acquisition should implement method evaluate")
+  def scalar_evaluate(self, x: np.ndarray) -> float:
+    assert len(x.shape) == 1, f"scalar_evaluate intended for use with vector inputs and not arrays of vector values x"
+    return self.evaluate(np.atleast_2d(x))[0][0]
```
Comment (nychiang) [inline src/hiopbbpy/opt/acquisition.py:24]: can [0][0] be replaced by .flatten()?

**Review by nychiang — APPROVED** [review]
LGTM --- just one simple question

File: src/hiopbbpy/opt/acquisition.py (hunk 1 above)
Comment (thartland) [reply src/hiopbbpy/opt/acquisition.py:24]: No. The intent of `scalar_evaluate` is to output a `float` the output of  `.flatten` will be an array whose extra array dimensions have been collapsed.
